### PR TITLE
Use task_id to get task_history

### PR DESCRIPTION
**Pull request by Sujanadh**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation
- [X] 🧑‍💻 Refactor
- [ ] ✅ Test
- [ ] 🤖 Build or CI
- [ ] ❓ Other (please specify)

## Related Issue

The latest updated task status was at the bottom of the list in the response.

## Describe this PR

This PR removes the param project_id and uses task_id in the path, which directly fetches the task history of that particular task, returning a response in the order of the latest update, which is descending order by id of taskhistory.

## Screenshots

```
[
  {
    "id": 151,
    "action_text": "Status changed from MAPPED to LOCKED_FOR_VALIDATION by: svcfmtm",
    "action_date": "2024-04-10T04:00:12.248890",
    "username": "svcfmtm",
    "profile_img": null,
    "status": "LOCKED_FOR_VALIDATION"
  },
  {
    "id": 150,
    "action_text": "Status changed from LOCKED_FOR_MAPPING to MAPPED by: svcfmtm",
    "action_date": "2024-04-10T03:59:22.988060",
    "username": "svcfmtm",
    "profile_img": null,
    "status": "MAPPED"
  },
  {
    "id": 134,
    "action_text": "Status changed from READY to LOCKED_FOR_MAPPING by: svcfmtm",
    "action_date": "2024-04-04T04:19:31.093732",
    "username": "svcfmtm",
    "profile_img": null,
    "status": "LOCKED_FOR_MAPPING"
  }
]
```

## Alternative Approaches Considered

Did you attempt any other approaches that are not documented in code?

## Review Guide

Notes for the reviewer. How to test this change?

## Checklist before requesting a review

- 📖 Read the FMTM Contributing Guide: <https://github.com/hotosm/fmtm/blob/main/CONTRIBUTING.md>
- 📖 Read the HOT Code of Conduct: <https://docs.hotosm.org/code-of-conduct>
- 👷‍♀️ Create small PRs. In most cases, this will be possible.
- ✅ Provide tests for your changes.
- 📝 Use descriptive commit messages.
- 📗 Update any related documentation and include any relevant screenshots.

### CODE REVIEW

**Review by NSUWAL123 — APPROVED** [review]

**Comment by spwoodcock** [comment]
I will merge this, but I am assuming this also probably requires frontend changes @NSUWAL123 to add the `?task_id=xxx` url param to the request 👍

File: src/backend/app/tasks/tasks_crud.py
Comment (spwoodcock) [inline src/backend/app/tasks/tasks_crud.py:339]: Related to https://github.com/hotosm/fmtm/issues/1326

Is it possible to keep both `project_id` and `task_id` params here, as in the future we want to be using the task id in relation to the project?

i.e. instead of `task_id=13879435` we want to use `project_id=22&task_id=10`

If needed for now we could pass a combination: `project_id=22&task_id=13879435`, where only the task id is used in the SQL. But we may refactor to use the project_id/task_id combo in the future & keeping both would make that easier.

**Review by spwoodcock — CHANGES_REQUESTED** [review]

**Comment by NSUWAL123** [comment]
> I will merge this, but I am assuming this also probably requires frontend changes @NSUWAL123 to add the `?task_id=xxx` url param to the request 👍

Yes @spwoodcock . I will update the frontend part  after this PR gets merged. Also the frontend has too many old code related to displaying and updating the task history that needs to be removed and updated as we are now following different approach.

File: src/backend/app/tasks/tasks_crud.py
Comment (Sujanadh) [reply src/backend/app/tasks/tasks_crud.py:339]: Yes, great, Thanks for pointing it out.

**Comment by Sujanadh** [comment]
- [X] added back project_id param in path

**Pull request merged**
